src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -90,6 +92,59 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
     <% if (request.getSession().getAttribute("user") != null) { %>
     <form action="/chat/<%= conversation.getTitle() %>" method="POST">
         <input type="text" name="message">
+        
+        <!-- Button trigger modal -->
+		<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModalLong">
+		  Send a GIF
+		</button>
+		
+		<!-- Modal -->
+		<div class="modal fade" id="exampleModalLong" tabindex="-1" role="dialog" aria-labelledby="exampleModalLongTitle" aria-hidden="true">
```

Comment:
Nit: This code would be easier to read in the GitHub UI if you re-wrapped the long lines.